context/context.go

```diff
@@ -43,6 +46,21 @@ func initOpVersion() {
 	OpVersion = MaxOpVersion
 }
 
+func initEtcdClient() error {
+	Hostname, err := os.Hostname()
```

Comment:
You can't use it without the initialization is the conclusion we have right? And more over if we are going with the IP based approach what I suggested in another comment, this is no more a valid case.